roles/photoprism/tasks/main.yml

```diff
@@ -0,0 +1,32 @@
+---
+- name: Create config directory
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: 0777
+  with_items:
+    - "{{ photoprism_data_directory }}/storage"
+
+- name: Install Docker container
+  docker_container:
+    name: photoprism
+    image: photoprism/photoprism
+    pull: true
+    ports:
+      - "{{ photoprism_port }}:2342"
+    volumes:
+      - "/etc/localtime:/etc/localtime:ro"
+      - "{{ photoprism_data_directory }}/storage:/photoprism/storage:rw"
+      - "{{ photoprism_photos_directory }}:/photoprism/originals:rw"
+    env:
+      PHOTOPRISM_SITE_URL: "https://photoprism.{{ ansible_nas_domain }}"
+      PHOTOPRISM_ADMIN_PASSWORD: "insecure"
+    restart_policy: unless-stopped
+    memory: 4g
+    labels:
+      traefik.enable: "{{ photoprism_available_externally  }}"
+      traefik.http.routers.gitlab.rule: "Host(`photoprism.{{ ansible_nas_domain }}`)"
```

Comment:
The traefik rules have `gitlab` instead of `photoprism` currently, this needs to be corrected

Reply:
There is also no entry in `nas.yml` so ansible doesn't know about the `photoprism` task.

Reply:
To add to `nas.yml`:

```
    - role: photoprism
      tags:
        - photoprism
      when: (photoprism_enabled | default(False))
 ```